src/test/codewhisperer/service/recommendationHandler.test.ts

```diff
@@ -141,11 +142,10 @@ describe('recommendationHandler', function () {
             assertTelemetry({
                 codewhispererRequestId: 'test_request',
                 codewhispererSessionId: 'test_request',
-                codewhispererLastSuggestionIndex: -1,
+                codewhispererLastSuggestionIndex: 1,
```

Comment:
i might missed the context, Andrew could you point me out where is the change that requires this change

Reply:
It's the movement of this part of code `recordServiceInvocationTelemetry`. It's moved after `this.recommendations` are updated. If we don't do that, a serviceInvocation will use an empty `this.recommendations` value and always send a `-1` in the first serviceInvocation event for the session, which is incorrect. This value in the first event should record the # of suggestions received so far minus one. (including the first response).

Reply:
sg